R/functions.R

```diff
@@ -242,6 +242,34 @@ get_biomart <- function(count_df, synid, version, host, filters, organism) {
     return(biomart_results)
   }
 }
+
+
+#'Check for SampleID Inconsistencies in Counts Matrix and Metadata
+#'
+#'Subsequent steps assume unique SampleIDs are present within both the counts matrix
+#'and the metadata. This function checks for duplicated SampleIDs in both the counts
+#'matrix and metadata before then ensuring each sample in the counts matrix has a 
+#'corresponding line in the metadata, and vice versa. 
+#'@inheritParams get_biomart
+#'@param md A data frame with sample identifiers as row names and relevant experimental covariates.
+#'@export
+check_mismatch <- function(md, count_df) {
+  # Isolate the SampleID's from the metadata and counts matrix
+  md_sampleid <- rownames(md)
+  cnt_sampleid <- colnames(count_df)
+  
+  # First check if these are duplicates in either of these. Returns an error if duplicates are found.
+  if (length(unique(md_sampleid)) != length(md_sampleid) ){
+    stop("ERROR: Unique sample IDs are required for subsequent functions. Please check your metadata input.")}
+  else if (length(unique(cnt_sampleid)) != length(cnt_sampleid)){
+    stop("ERROR: Unique sample IDs are required for subsequent functions.Please check your counts input")}
+  
+  # Ensures each sample in the metadata has a corresponding column in the counts matrix and vice versa
+  else if ((length(setdiff(md_sampleid, cnt_sampleid)) > 0) || (length(setdiff(cnt_sampleid, md_sampleid)) > 0)){
+    stop("ERROR: All SampleIDs provided must exist both in the counts and metadata inputs")
```

Comment:
I think it could be clearer to say: "All sample identifiers in the counts matrix and metadata file must match."

Also, you don't need to type "ERROR" as `stop()` will append it automatically: 

``` r
x = 3

if (x > 1) {stop("stop function")}
#> Error in eval(expr, envir, enclos): stop function
```

<sup>Created on 2021-07-30 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>